aioshelly/rpc_device/wsrpc.py

```diff
@@ -261,7 +261,11 @@ def _pong_not_received(self) -> None:
 
     async def disconnect(self) -> None:
         """Disconnect all sessions."""
-        self._rx_task = None
+        if self._rx_task is not None:
+            self._rx_task.cancel()
+            with contextlib.suppress(asyncio.CancelledError):
+                await self._rx_task
+            self._rx_task = None
```

Comment:
This is likely the problem as we didn't cancel the task on disconnect. But cancelling it isn't safe unless we wrap it in finally to ensure it still cleans up